packages/nimble-components/src/banner/styles.ts

```diff
@@ -0,0 +1,241 @@
+        margin-top: 8px;
+    }
+
+    .dismiss nimble-button {
+        height: 16px;
+        width: 16px;
+        ${iconSize.cssCustomProperty}: 14px;
+        ${buttonLabelFontColor.cssCustomProperty}: ${White};
+        ${borderHoverColor.cssCustomProperty}: transparent;
+        ${fillSelectedColor.cssCustomProperty}: ${hexToRgbaCssColor(
+            White,
+            0.2
+        )};
+    }
+
+    .dismiss nimble-button:focus-within {
+        outline: 2px solid ${White};
+    }
+
+    .dismiss nimble-button:hover {
+        background: ${hexToRgbaCssColor(White, 0.2)};
+    }
+`.withBehaviors(
+    themeBehavior(
+        Theme.light,
+        css`
+            :host {
+                background: ${Black75};
+            }
+        `
+    ),
+    themeBehavior(
+        Theme.dark,
+        css`
+            :host {
+                background: ${Black75};
+            }
+        `
+    ),
+    themeBehavior(
+        Theme.color,
+        css`
+            :host {
+                background: ${applicationBackgroundColor};
+            }
+
+            .container {
+                background: ${hexToRgbaCssColor(White, 0.3)};
+            }
+        `
+    ),
+    new MultivaluePropertyStyleSheetBehavior(
+        'severity',
+        BannerSeverity.error,
+        css``.withBehaviors(
+            themeBehavior(
+                Theme.light,
+                css`
+                    :host {
+                        background: ${Fail100LightUi};
+                    }
+                `
+            ),
+            themeBehavior(
+                Theme.dark,
+                css`
+                    :host {
+                        background: ${BannerFail100DarkUi};
+                    }
+                `
+            )
+        )
+    ),
+    new MultivaluePropertyStyleSheetBehavior(
+        'severity',
+        BannerSeverity.warning,
+        css``.withBehaviors(
+            themeBehavior(
+                Theme.light,
+                css`
+                    :host {
+                        background: ${Warning100LightUi};
+                    }
+                `
+            ),
+            themeBehavior(
+                Theme.dark,
+                css`
+                    :host {
+                        background: ${Warning100DarkUi};
+                    }
+                `
+            )
+        )
+    ),
+    new MultivaluePropertyStyleSheetBehavior(
+        'severity',
+        BannerSeverity.info,
+        css``.withBehaviors(
+            themeBehavior(
+                Theme.light,
+                css`
+                    :host {
+                        background: ${Information100LightUi};
+                    }
+                `
+            ),
+            themeBehavior(
+                Theme.dark,
+                css`
+                    :host {
+                        background: ${Information100DarkUi};
+                    }
+                `
+            )
+        )
+    )
+);
```

Comment:
I'd recommend reviewing the responding to attribute values section of the docs: https://github.com/ni/nimble/blob/main/packages/nimble-components/CONTRIBUTING.md#responding-to-attribute-values

Particularly the guidance:

> Behaviors are useful when a large block of styles is overridden based on the attibute configuration, i.e. on the order of replacing a large chunk of the stylesheet based on the configuration.

The reasoning is two fold:
- performance: there is a performance cost both for large stylesheets and a large number of stylesheets. If a stylesheet is large and has a lot of dead styles not relevant to a specific element config it's extra work for the browser. If there are a large number of stylesheets (rouchly the number of `css` tags that end up being attached per element), there is a large cascade cost.
- maintenance: we can break-up the styles into chunks and completely swap them out. Makes it easier to evaluate the styles because you only need to be aware of a smaller set at a time.

So I think the current organization isn't helping with perf or maintenance. We are creating a lot of small stylesheets and the above is very difficult to understand.

I'd recommend just having the theme layer as behaviors and use attribute selectors for the severity states.

ie.

```ts
css`
:host {}
`.withBehaviors(
  themeBehavior(Theme.light, css`
    :host {}
    :host[severity='warning'] {}
  `), etc.
```

Reply:
Makes sense. Changed.